ds3-sdk/src/main/java/com/spectralogic/ds3client/Ds3InputStreamEntity.java

```diff
@@ -41,6 +45,14 @@ public long getBufferSize() {
 
     @Override
     public void writeTo(final OutputStream outStream) throws IOException {
-        IOUtils.copy(this.getContent(), outStream, bufferSize);
+        final long startTime = PerformanceUtils.getCurrentTime();
+        final long totalBytes = IOUtils.copy(this.getContent(), outStream, bufferSize);
+        final long endTime = PerformanceUtils.getCurrentTime();
+
+        if (this.getContentLength() != -1 && totalBytes != this.getContentLength()) {
```

Comment:
Have we confirmed that the Apache HTTP client will return -1 when the content length is not set?